common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -333,6 +345,146 @@ public StyledText prependPart(StyledTextPart part) {
         return new StyledText(newParts, temporaryWorkaround, clickEvents, hoverEvents);
     }
 
+    /**
+     * Splits this {@link StyledText} into multiple {@link StyledText}s at the given index.
+     * <p> Note that {@link PartStyle.StyleType.NONE} is used when splitting.
+     * @param regex the regex to split at
+     * @return the split {@link StyledText}s
+     */
+    public StyledText[] split(String regex) {
```

Comment:
This split method doesn't actually work like the String split method: when splitting a string it omits empty strings if they are at the last index of the result so if you have something like this "foo\n\n§0bar" splitting on "\n" this method would return an array of size 2(or 3 idk) and the string method one of size 4. This would be an issue in QuestInfo.